### PR TITLE
Align `regenerator-transform` import with native ESM

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | 
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

I missed this in https://github.com/babel/babel/pull/12795 because it was using `export ... from` rather than import.

`regenerator-transform` is a compiled CJS module, so the `default` export in Node.js is `module.exports` and we'll need to get `.default` from `module.exports` (thus `.default` from the `default` export).

### CODE REVIEW

File: babel.config.js [hunk 1]
```diff
@@ -475,15 +475,14 @@ function pluginNodeImportInteropBabel({ template }) {
 }
 
 function pluginNodeImportInteropRollup({ types: t }) {
-  const depsUsing__esModuleAndDefaultExport = [
-    src => src.startsWith("babel-plugin-polyfill-"),
-  ];
+  const depsUsing__esModuleAndDefaultExport = src =>
+    src.startsWith("babel-plugin-polyfill-") || src === "regenerator-transform";
```
Comment (hzoo) [inline babel.config.js:479]: Just wondering, is there anything else other than `regenerator-transform` that would need to be added? Would we need to update this if we added another similar dep?

Oh so via https://github.com/babel/babel/pull/13017, it's only esm/cjs packages that have been transformed with babel?

Comment (nicolo-ribaudo) [reply babel.config.js:479]: Yes, the only problem is dependencies that have been transformed with Babel.

I managed to create a working build just with this change so I _think_ it's the only one (also I tried looking for other `export ... from` of dependencies and this is the only one), but if we'll ever find a new one we can always add it to this list.

Comment (hzoo) [reply babel.config.js:479]: Ok was just worried it would break without us knowing and whether one could verify!

**Review by hzoo — APPROVED** [review]

**Comment by nicolo-ribaudo** [comment]
Merging since this is a really small internal change, that I forgot to do in another PR.

The new tests doesn't test changed behavior, but it's just to be 100% sure I didn't accidentally break something.

**Pull request merged**